packages/errors/src/index.ts

```diff
@@ -55,8 +62,9 @@ export const isWorkflowError = (error: Error) => error.name.includes('WorkflowEr
 /**
  * Returns true if the error is a DynamoDB conditional check exception.
  */
-export const isConditionalCheckException = (error: ErrorWithOptionalCode) =>
-  error.name === 'ConditionalCheckFailedException' || error.code === 'ConditionalCheckFailedException';
+export const isConditionalCheckException = (error: any) => {
+  return error.name === "ConditionalCheckFailedException";
```

Comment:
This one appears to be a special case. The service throwing the error is DynamoDBDocument, which is from `@aws-sdk/lib-dynamodb`. That might work slightly differently. The error does appear to have the correct class inherited but `instanceof` does not work here where it does elsewhere.